### ISSUE TITLE
Share / Export bookmarks

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#189]
From user:

> ability to Share/ Export Bookmarks

**Comment by huperniketes** [comment]
I'm planning to add support for a number of formats for exporting as well as importing (where feasible). Those formats include CSV, Word, Excel, OpenOffice, and PDF.

**Comment by tuomas2** [comment]
I would go with only CSV first.

**Comment by tuomas2** [comment]
Tried but I don't know why I can't assign this to @huperniketes

**Comment by huperniketes** [comment]
Ha! I can't assign any issues either.

I'll start with CSV. It's a different library anyhow.

Is there a TDD framework in place for the app?

**Comment by tuomas2** [comment]
Rough estimate: couple of hours of working time for CSV export.

**Comment by KomiB-Atsou** [comment]
Please is there any known pull request about this issue?

**Comment by tuomas2** [comment]
Nope. Exporting studypads (label + associated bookmarks and notes) is implemented but only in Andbible's internal (sqlite) format, not CSV etc.

**Comment by GWev** [comment]
Dear Tuomas, a while ago you've told me the development of a feature that allows to export correctly all bookmarks and notes was not a priority, but doable if there is a sponsor, taking approximately a couple of hours.
According to this post https://andbible.org/2025/05/17/matching-contribution-opportunity-for-andbible-development/ there is a chance to let it happen. I would like to sponsor what is needed - if possible with this donation partnership - if you agree and could send me the needed details and amount ! Thank you!

**Comment by tuomas2** [comment]
Unfortunately this ticket is not included in matching contribution offer. 
My new estimate for this is 4h, so if we get 4h contribution for this ticket [via my webshop](https://shop.tuomasairaksinen.fi/product/specific-feature-development-for-andbible-discount-12-5), it will be done.

**Comment by tuomas2** [comment]
Sorry @GWev there's some confusion about this issue. I will check and come back to this later.

**Comment by GWev** [comment]
Thank you Tuomas. I will wait for instructions. Could you pls check with the "contribution offer", if it could be possibly accepted as "matching" The whole 4h sponsoring would be tricky for me, but when you tell me the final decision pls let me know and I will see how to handle it. Thanx

**Comment by tuomas2** [comment]
Okay, clarified this too now. I had discussion and this can be included in matching contribution too.
Based on matching contribution, 2 hour contribution via [my webshop](https://shop.tuomasairaksinen.fi/product/specific-feature-development-for-andbible-discount-12-5) will make it happen. Please mark this ticket (#189) as a comment in your order. Thank you!

**Comment by tuomas2** [comment]
Implemented in #3445

**Comment by tuomas2** [comment]
This is now included in [build 882](https://github.com/AndBible/and-bible/releases/tag/production-882). Testing now is appreciated!

Some [preliminary documentation](https://github.com/AndBible/and-bible/blob/develop/docs/csv-import-export.md)

**Comment by tuomas2** [comment]
Please let me know if you have tested the feature (import and/or export). It is going to production soon.

**Comment by GWev** [comment]
I cannot test anything at the moment. because it is not published on F-Droid (no g00gl on my phone). But I will ASAP. Thank you !

Am 19. Juli 2025 08:48:52 UTC schrieb Tuomas Airaksinen ***@***.***>:
>tuomas2 left a comment (AndBible/and-bible#189)
>
>Please let me know if you have tested the feature. It is going to production soon.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/AndBible/and-bible/issues/189#issuecomment-3092178437
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Jean
www.gateway-ev.de

**Comment by tuomas2** [comment]
Okay triggering F-droid build now. It should appear there in 1-2 days.
